src/arrayMethodReduce.js

```diff
@@ -5,7 +5,19 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let startValue = initialValue;
+    let startIndex = 0;
+
+    if (arguments.length < 2) {
+      startValue = this[0];
+      startIndex = 1;
+    }
```

Comment:
It's a good case to use ternary here. 

If you don't know how to implement it - ask for help in fe_chat